### PR TITLE
Update deps: efel5*, pytest 8, python38+  vcs: #minor

### PR DESCRIPTION
**Pull request by anilbey**
addresses #100. Removes nose test style setup funcs as well.

### CODE REVIEW

**Review by AurelienJaquier — APPROVED** [review]

**Comment by anilbey** [comment]
@AurelienJaquier more changes are added since your last review. Could you check again? Thanks :)

File: tests/unit_tests/test_GUI_simulator_class.py [hunk 1]
```diff
@@ -23,11 +23,10 @@
 example_dir = os.path.join("examples", "sscx_sample_dir")
 
 
-class TestNeuronSimulation(object):
-
+class TestNeuronSimulation:
     """Test class for NeuronSimulation"""
 
-    def setup(self):
+    def setup_method(self):
```
[the remark this answers is not in the document]
Comment (anilbey) [reply tests/unit_tests/test_GUI_simulator_class.py:29]: this is the breaking pytest change. https://pytest.org/en/8.0.x/deprecations.html#support-for-tests-written-for-nose

Pytest 8 does not accept that style of setup anymore.

File: examples/thalamus_sample_dir/config/features/dNAD_ltb.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
       {"feature": "AP1_amp", "val": [45.8336, 4.5541], "n": 12, "fid": 690, "strict_stim": true},
       {"feature": "AP2_amp", "val": [43.3136, 6.5238], "n": 12, "fid": 872, "strict_stim": true},
       {"feature": "time_to_first_spike", "val": [43.3185, 13.6207], "n": 12, "fid": 534, "strict_stim": true},
-      {"feature": "Spikecount", "val": [4.5702, 1.8613], "n": 12, "fid": 716, "strict_stim": true},
+      {"feature": "", "val": [4.5702, 1.8613], "n": 12, "fid": 716, "strict_stim": true},
```
[the remark this answers is not in the document]
Comment (anilbey) [reply examples/thalamus_sample_dir/config/features/dNAD_ltb.json:35]: No definitely not, let me fix it.

File: examples/thalamus_sample_dir/config/features/dNAD_ltb.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
   "RMP": {
     "soma.v": [
       {"feature": "steady_state_voltage_stimend", "val": [-78.2502, 3.4762], "n": 10, "fid": 1109, "strict_stim": true},
-      {"feature": "Spikecount_stimint", "val": [0.0, 0.001], "n": 10, "fid": 1114, "strict_stim": true}
+      {"feature": "_stimint", "val": [0.0, 0.001], "n": 10, "fid": 1114, "strict_stim": true}
```
[the remark this answers is not in the document]
Comment (anilbey) [reply examples/thalamus_sample_dir/config/features/dNAD_ltb.json]: Thanks! It is fixed in this push

**Comment by AurelienJaquier** [comment]
Looks great, thanks!

**Pull request merged**